bloom/consumed.py

```diff
@@ -0,0 +1,123 @@
+
+    Further, please note that RecentlyConsumed is neither thread-safe nor safe
+    to use across instances.  Please lock around your RecentlyConsumed
+    instances/accesses.  A good choice for such a lock is .memlock.MemLock.
+    '''
+
+    _RANDOM_KEY_PREFIX = 'consumed:'
+    _MAXLEN = 1000
+
+    def __init__(self, memcache=None, key=None, maxlen=_MAXLEN):
+        super(RecentlyConsumed, self).__init__(memcache=memcache, key=key)
+        self._maxlen = maxlen
+        self._load_from_memcache()
+
+    @property
+    def maxlen(self):
+        return self._maxlen
+
+    @maxlen.setter
+    def maxlen(self, value):
+        message = "attribute 'maxlen' of '{}' objects is not writable"
+        message = message.format(self.__class__.__name__)
+        raise AttributeError(message)
+
+    def _load_from_memcache(self):
+        string = self.memcache.get(self.key, default='[]')
+        list_ = json.loads(string)
+        if self.maxlen is not None and len(list_) > self.maxlen:
+            message = 'persistent {} beyond its maximum size'
+            message = message.format(self.__class__.__name__)
+            raise IndexError(message)
+        self._deque = collections.deque(list_)
+        self._set = set(list_)
+
+    def _store_to_memcache(self):
+        if self._set:
+            list_ = list(self._deque)
+            string = json.dumps(list_)
+            self.memcache.set(self.key, string, noreply=True)
+        else:
+            self.memcache.delete(self.key)
+
+    def _prune_to_maxlen(self):
+        while self.maxlen is not None and len(self) > self.maxlen:
+            value = self._deque.popleft()
+            self._set.remove(value)
+
+    def __len__(self):
+        'Return the number of items in the RecentlyConsumed.'
+        return len(self._set)
+
+    def __contains__(self, value):
+        'rc.__contains__(element) <==> element in rc.'
+        return unicode(value) in self._set
+
+    def append(self, value):
+        'Add an element to the right side of the RecentlyConsumed.'
+        value = unicode(value)
+        if value not in self:
+            self._deque.append(value)
+            self._set.add(value)
+            self._prune_to_maxlen()
+            self._store_to_memcache()
+
+    def extend(self, values):
+        'Extend a RecentlyConsumed by appending elements from the iterable.'
+        values = tuple(unicode(value) for value in values if value not in self)
+        if values:
+            self._deque.extend(values)
+            self._set.update(values)
+            self._prune_to_maxlen()
+            self._store_to_memcache()
+
+    def clear(self):
+        'Remove all elements from the RecentlyConsumed.'
+        self._deque.clear()
+        self._set.clear()
+        self._store_to_memcache()
+
+    def __repr__(self):
+        'Return the string representation of the RecentlyConsumed data struct.'
+        repr = self.__class__.__name__ + '('
+        repr += str(list(self._deque))
```

Comment:
This could get a little verbose for normal purposes. What if this just includes `len(self._deque)` and we can separately get the full deque if we need to debug?

Reply:
Or `['foo', 'bar', ...]` if it gets past some threshold?